ss_admin_console/main/tests/classes/BaseDataFeederTest.cls

```diff
@@ -0,0 +1,136 @@
+            addKeyField(Contact.Email).addKeyField(Contact.LastName).addKeyField('Account.Name').addDataField(Contact.Email).
+            addDataField(Contact.LastName).addDataField(Contact.FirstName).addDataField('Account.Name').
+            addReferenceField(Contact.AccountId, new String[] {'Name'});
+
+
+    @IsTest 
+    private static void testFeedUpsertMocksNegative() {
+        fflib_ApexMocks mocks = new fflib_ApexMocks();
+        FeedLogger logger = (FeedLogger) mocks.mock(FeedLogger.class);
+        BaseDataFeeder df = baseDataFeederInstance(mocks, true, true);
+        Test.startTest();
+        df.feed(true, logger);
+        Test.stopTest();
+        ((FeedLogger) mocks.verify(logger, 1)).error(fflib_Match.stringStartsWith('FAILED'));
+    }
+
+    @IsTest 
+    private static void testFeedUpsertMocksPositive() {
+        fflib_ApexMocks mocks = new fflib_ApexMocks();
+        FeedLogger logger = (FeedLogger) mocks.mock(FeedLogger.class);
+        BaseDataFeeder df = baseDataFeederInstance(mocks, false, true);
+        Test.startTest();
+        df.feed(true, logger);
+        Test.stopTest();
+        ((FeedLogger) mocks.verify(logger, 1)).log('Contact', '2 records were proceed.', SSACConstants.UPSERTED_STATUS);
+    }
+
+    @IsTest 
+    private static void testFeedInsertMocksNegative() {
+        fflib_ApexMocks mocks = new fflib_ApexMocks();
+        FeedLogger logger = (FeedLogger) mocks.mock(FeedLogger.class);
+        BaseDataFeeder df = baseDataFeederInstance(mocks, true, true);
+        Test.startTest();
+        df.feed(false, logger);
+        Test.stopTest();
+        ((FeedLogger) mocks.verify(logger, 1)).error(fflib_Match.stringStartsWith('FAILED'));
+    }
+
+    @IsTest 
+    private static void testFeedInsertMocksPositive() {
+        fflib_ApexMocks mocks = new fflib_ApexMocks();
+        FeedLogger logger = (FeedLogger) mocks.mock(FeedLogger.class);
+        BaseDataFeeder df = baseDataFeederInstance(mocks, false, true);
+        Test.startTest();
+        df.feed(false, logger);
+        Test.stopTest();
+        ((FeedLogger) mocks.verify(logger, 1)).log('Contact', '1 records were proceed.', SSACConstants.INSERTED_STATUS);
+    }
+
+    @IsTest 
+    private static void testFeedInsertMocksPositiveAccounts() {
+        fflib_ApexMocks mocks = new fflib_ApexMocks();
+        FeedLogger logger = (FeedLogger) mocks.mock(FeedLogger.class);
+        BaseDataFeeder df = baseDataFeederInstance(mocks, false, false);
+        Test.startTest();
+        df.feed(false, logger);
+        Test.stopTest();
+        ((FeedLogger) mocks.verify(logger, 1)).log('Account', '1 records were proceed.', SSACConstants.INSERTED_STATUS);
+    }
+    
+    private static BaseDataFeeder baseDataFeederInstance(fflib_ApexMocks mocks, Boolean clearResults, Boolean isChild) {
+        DataFilter filter = (DataFilter) mocks.mock(DataFilter.class);
+        mocks.startStubbing();
+        if (clearResults) {
+            if (isChild) {
```

Comment:
Instead of clearing pre-populated list, you can use a new empty list.
e.g.: 
if (clearResults) {
    mocks.when(filter.filter()).thenReturn(new List<SSACResult>());
} else {
     ...
}